common_content_filter/test/test_api.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <test_msgs/msg/basic_types.h>
+#include <osrf_testing_tools_cpp/scope_exit.hpp>
+#include <rmw/rmw.h>
+#include <rmw/serialized_message.h>
+
+#include "common_content_filter/api.h"
+
+class TestCommonContentFilterAPI : public ::testing::Test
```

Comment:
About other tests except API.  
The main logic of using PEGTL is from eProsima's code. So I think these tests are unnecessary. The modified point is to get correct value from deserialized message.  
So test focus on test this.  
Maybe we need to construct a complex message for test
e.g.  
level1_b.level2_c[2].level3_e[5]
At level3, it includes all kinds of basic type. Test is to get each value of each item in level3.  
What do you think ?

Reply:
OK, updated.